File: tests/fixtures/python_simple/routes/payments.py
from auth.middleware import auth_required
from models.user import get_user_by_id
from models.payment import create_payment, get_payments_by_user
from services.stripe import charge_customer

@auth_required
def create_payment_endpoint(data: dict, current_user: dict = None):
    """POST /payments"""
    user = get_user_by_id(current_user["user_id"])
    payment = create_payment(user, data["amount"])
    charge_result = charge_customer(user, payment)
    return {"payment": payment, "charge": charge_result}

@auth_required
def get_user_payments(current_user: dict = None):
    """GET /payments"""
    user = get_user_by_id(current_user["user_id"])
    payments = get_payments_by_user(user)
    return {"payments": payments}

File: tests/fixtures/typescript_simple/src/routes/payments.ts
import { authMiddleware } from "../auth/middleware";
import { makeUser } from "../types/user";
import { makePayment, PaymentType } from "../types/payment";
import { chargeCustomer } from "../services/stripe";

export function createPaymentEndpoint(req: { token: string; amount: number }): { payment: PaymentType } | { error: string } {
  const payload = authMiddleware(req);
  if (payload === null) return { error: "Unauthorized" };
  const user = makeUser(payload.userId, "user@example.com");
  const payment = makePayment(user, req.amount);
  chargeCustomer(user, payment);
  return { payment };
}
